scenario_player/main.py

```diff
@@ -269,12 +269,13 @@ def _load_environment(environment_file: IO) -> EnvironmentConfig:
     environment = json.load(environment_file)
     assert isinstance(environment, dict)
 
-    matrix_server_list = environment.pop("matrix_server_list", None)
-    matrix_servers: Sequence[Union[URI, Literal["auto"]]] = ["auto"]
-    if matrix_server_list is not None:
-        matrix_servers = [URI(uri) for uri in get_matrix_servers(matrix_server_list)]
-        if len(matrix_servers) < 4:
-            matrix_servers = list(islice(cycle(matrix_servers), 4))
+    matrix_server_list = environment.pop(
+        "matrix_server_list",
+        DEFAULT_MATRIX_KNOWN_SERVERS[Environment(environment["environment_type"])],
+    )
+    matrix_servers: Sequence[URI] = get_matrix_servers(matrix_server_list)  # type: ignore
+    if len(matrix_servers) < 4:
```

Comment:
Where is the 4 coming from?

Reply:
This is the highest number of servers any existing scenario requires. Neither elegant nor flexible, but works for now. This is also not new in this PR.